lib/floe/workflow/runner/podman.rb

```diff
@@ -11,17 +11,17 @@ def initialize(*)
           super
         end
 
-        def run!(resource, env = {}, secrets = {})
+        def run_async!(resource, env = {}, secrets = {})
           raise ArgumentError, "Invalid resource" unless resource&.start_with?("docker://")
 
           image = resource.sub("docker://", "")
 
-          params = ["run", :rm]
+          params = ["run", :detach]
           params += env.map { |k, v| [:e, "#{k}=#{v}"] } if env
 
           if secrets && !secrets.empty?
```

Comment:
to make codeclimate happy, probably want to add something like `add_secrets`
Wasn't sure if `create_params` would make sense as well, but this seems simpler

```ruby
def add_secrets(params, secrets)
  return params unless secrets && !secrets.empty?

  ...

  params << [:e, "SECRETS=/run/secrets/#{secret_guid}"]
  params << [:secret, secret_guid]

  params
end
```

Reply:
I wasn't too worried about codeclimate because the `run!` methods before were also over the cognitive complexity limit by the same amount so these aren't _really_ new (except the docker one)